### ISSUE TITLE
routeRules prerender not work

### ISSUE THREAD

**Issue by a1mersnow** [nuxt/nuxt#22661]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.20.1`
- Nuxt Version:     `3.6.5`
- Nitro Version:    `2.5.2`
- Package Manager:  `pnpm@8.6.12`
- Builder:          `vite`
- User Config:      `app`, `ui`, `devtools`, `typescript`, `vue`, `modules`, `content`, `colorMode`, `vite`
- Runtime Modules:  `@nuxt/content@2.7.2`, `@nuxthq/ui@2.7.0-28200741.b8b7d74`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-2mnq8w-9arzti?file=nuxt.config.ts

### Describe the bug

There are no `.html` files in `.output` dir.

**Comment by danielroe** [comment]
Check `.output/public/index.html`.

**Comment by a1mersnow** [comment]
@danielroe But, there is not any `.html` file.

**Comment by a1mersnow** [comment]
@danielroe I found the problem! The config below doesn't generate any html files:
```js
routeRules: {
  '/**': { prerender: true },
},
```

So I guess the `prerender` doesn't support glob pattern, right?

So is there any solution for this(using glob pattern for prerender)?

**Comment by danielroe** [comment]
If you pass a glob pattern, then it will prerender any routes _matching_ that glob pattern that are discovered whilst crawling your HTML pages. But it doesn't know to prerender the root path. So additionally you will need to set `nitro.prerender.routes` to `['/']`.

**Comment by richgcook** [comment]
Sorry to bring this back up (but better than making a new issue, right?) but the following routeRules doesn't prerender any routes – is this sort of approach not possible?

```
routeRules: {
	// Prerender all routes
	'/**': { prerender: true },
	// Exclude API routes from prerendering
	'/api/**': { prerender: false },
	// API routes that fetch Are.na content use SWR (cached for 1 hour)
	'/api/arena/**': { swr: 3600 },
}
```

**Comment by danielroe** [comment]
@richgcook You still need to pass a starting URL. Does my comment above make sense?

**Comment by richgcook** [comment]
@danielroe Thanks very much for the reply. Yeah, I thought this would've solved it but it only prerenders that one route.

```js
nitro: {
	prerender: {
		routes: ['/']
	}
},

routeRules: {
	// Prerender all routes
	'/**': { prerender: true },
	// Exclude API routes from prerendering
	'/api/**': { prerender: false },
	// API routes that fetch Are.na content use SWR (cached for 1 hour)
	'/api/arena/**': { swr: 3600 },

	// Redirects
	'/program/labs': { redirect: '/program' },
	'/community/profiles': { redirect: '/community' }
},
```

```console
ℹ Prerendering 1 routes                                                                                                                                                nitro 21:19:53
  ├─ / (2190ms)                                                                                                                                                         nitro 21:19:56
  ├─ /_payload.json (16ms)                                                                                                                                              nitro 21:19:56
ℹ Prerendered 2 routes in 11.709 seconds                                                                                                                               nitro 21:20:00
✔ Generated public .output/public   
```

**Comment by richgcook** [comment]
Wait, routeRules shouldn't be in the nitro object, right? routeRules should be a root level defineNuxtConfig object, correct?

**Comment by richgcook** [comment]
'/community/*': { prerender: true },` is not working for me either so I can't help but feel like I'm doing something wrong. Should I need to include anything else in order to get this to work?

**Comment by danielroe** [comment]
there is something to fix here - namely, we should inject the paths to prerender if you have `prerender: true` on any routeRule, and prerender at least one route.

at the moment we only inject paths from your `~/pages` directory if you have `crawlLinks: true` enabled. That could be a workaround for you at the moment.

**Comment by richgcook** [comment]
> there is something to fix here - namely, we should inject the paths to prerender if you have `prerender: true` on any routeRule, and prerender at least one route.

Yes, exactly. I thought it would've worked like this. Lmk what I can do to request this feature 👍🏻 

> at the moment we only inject paths from your `~/pages` directory if you have `crawlLinks: true` enabled. That could be a workaround for you at the moment.

I was reading about the difference between nuxt's routeRules and nitro's routeRules and they seem to work in parallel, with a few differences, so I can use either or for my routeRules. However, if I set nitro.prerender.crawlLinks to true then it of course creates html files of the routes but it ignores the routeRules e.g. if I set a specific page to render as swr or ssr then it's ignored and a static file is outputted for it regardless.

Thanks for all your help.

**Comment by alexcroox** [comment]
Is it possible to support the glob pattern with wildcard at the start of the path?

```
routeRules: {
    '/': { prerender: true, ssr: true },
    '/auth/**': { prerender: true, ssr: true },
    '/p/admin/**': { prerender: false, ssr: false }, <-- WORKS
    '/**/admin/**': { prerender: false, ssr: false }, <-- DOESN'T WORK
  },
```
